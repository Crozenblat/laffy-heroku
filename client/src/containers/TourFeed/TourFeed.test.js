import React from "react";
import {render, screen} from "@testing-library/react";
import '@testing-library/jest-dom/extend-expect';
import {BrowserRouter} from "react-router-dom";

import seed from "../../../backend/seed";

import TourFeed from "./TourFeed";

describe("Tour Date Container", () => {
    const userOne = seed.users[0];

    it("should render correctly", () => {
        let wrapper = render(
            <BrowserRouter>
                <TourFeed userId={userOne._id} comedian={null}/>            
            </BrowserRouter>
        );

        expect(wrapper).toMatchSnapshot();
    });

    it("should render tour dates correctly from multiple ccomedians", async () => {
        render(
            <BrowserRouter>
                <TourFeed userId={userOne._id} comedian={null}/>            
            </BrowserRouter>
        );
        let tourDates = await screen.findAllByTestId("tour-date");
        
        expect(tourDates).toHaveLength(2);

        test.each([
            [tourDates[0], "Hannibal Buress", "Miami", "FL", "New World Center", "December 8th", "2020", "Sunday", "8:30pm"],
            [tourDates[1], "Tig Notaro", "New York", "NY", "Apollo Theater", "November 17th", "2020", "Tuesday", "TBA"]
        ])("display a tour date for %s", (tourDate, comedian, city, state, venue, date, year, day, time) => {
            expect(tourDate).toContainElement(screen.getAllByAltText(comedian));
            expect(tourDate).toContainElement(screen.getByText(comedian));
            expect(tourDate).toContainElement(screen.getByText(city));
            expect(tourDate).toContainElement(screen.getByText(state));
            expect(tourDate).toContainElement(screen.getByText(venue));
            expect(tourDate).toContainElement(screen.getByText(date));
            expect(tourDate).toContainElement(screen.getByText(year));
            expect(tourDate).toContainElement(screen.getByText(day));
            expect(tourDate).toContainElement(screen.getByText(time));
            expect(tourDate).toContainElement(screen.getByRole("button", {name: "Get Tickets!"}));
        });
    });

    it("should display an alert when a tour date has been announced recently", async () => {
        render(
            <BrowserRouter>
                <TourFeed userId={userOne._id} comedian={null}/>
            </BrowserRouter>
        );
        let tourDates = await screen.findAllByTestId("tour-date");

        expect(tourDates[0]).toContainElement(screen.getByText("Hannibal Buress"));
        expect(tourDates[0]).toContainElement(screen.getByText("New Date!"));

        expect(tourDates[1]).toContainElement(screen.getByText("Tig Notaro"));
        expect(tourDates[1]).not.toContainElement(screen.getByText("New Date!"));

    });

    it("should correctly render upcoming shows from a single comedian", async () => {
        render(
            <BrowserRouter>
                <TourFeed userId={null} comedian={seed.comedians[0].name}/>
            </BrowserRouter>
        );
        let tourDates = await screen.findAllByTestId("tour-date");
        
        expect(tourDates).toHaveLength(3);

        test.each([
            [tourDates[0], "Jerry Seinfeld", "Tampa", "FL", "Straz Center", "January 26th", "2021", "Tuesday", "8:30pm"],
            [tourDates[1], "Jerry Seinfeld", "Los Angeles", "CA", "The Comedy Store", "TBD", "2020"],
            [tourDates[2], "Jerry Seinfeld", "New York", "NY", "Radio City Music Hall", "TBA", "2020"]
        ])("display a tour date for %s", (tourDate, comedian, city, state, venue, date, year, day, time) => {
            expect(tourDate).toContainElement(screen.getAllByAltText(comedian));
            expect(tourDate).toContainElement(screen.getByText(comedian));
            expect(tourDate).toContainElement(screen.getByText(city));
            expect(tourDate).toContainElement(screen.getByText(state));
            expect(tourDate).toContainElement(screen.getByText(venue));
            expect(tourDate).toContainElement(screen.getByText(date));
            expect(tourDate).toContainElement(screen.getByText(year));
            expect(tourDate).toContainElement(screen.getByText(day));
            expect(tourDate).toContainElement(screen.getByText(time));
            expect(tourDate).toContainElement(screen.getByRole("button", {name: "Get Tickets!"}));
        });
    });

    it("should display a message when a comedian has no current shows scheduled", async () => {
        render(
            <BrowserRouter>
                <TourFeed userId={null} comedian={seed.comedians[0].name}/>
            </BrowserRouter>
        );
        let tourDates = await screen.findAllByTestId("tour-date");
        let message = screen.getByText("No tour dates currently scheduled.");

        expect(tourDates).toHaveLength(0);
        expect(screen.getByText(message)).toBeInTheDocument();
    });

    it("should display a message when there are no current shows coming near a user", async () => {
        render(
            <BrowserRouter>
                <TourFeed userId={userOne._id} comedian={null}/>
            </BrowserRouter>
        );
        let tourDates = await screen.findAllByTestId("tour-date");
        let message = screen.getByText("No tours currently coming near you.");

        expect(tourDates).toHaveLength(0);
        expect(screen.getByText(message)).toBeInTheDocument();
    });
});