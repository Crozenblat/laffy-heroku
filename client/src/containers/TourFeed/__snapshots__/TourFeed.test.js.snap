// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tour Date Container should render correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-AxhUy kNWbAH"
      >
        <div
          class="sc-AxheI jzdBSq"
        >
          <svg
            class="sc-AxgMl bCKtDA"
            viewBox="0 0 512 512"
            x="0px"
            y="0px"
          >
            <path
              d="M256,0C115.2,0,0,115.2,0,256s115.2,256,256,256s256-115.2,256-256S396.8,0,256,0z M256,51.2
 c28.16,0,48.64,23.04,46.08,51.2L281.6,307.2h-51.2l-20.48-204.8C207.36,74.24,227.84,51.2,256,51.2z M256,460.8
 c-28.16,0-51.2-23.04-51.2-51.2c0-28.16,23.04-51.2,51.2-51.2s51.2,23.04,51.2,51.2C307.2,437.76,284.16,460.8,256,460.8z"
            />
          </svg>
          New Date!
        </div>
        <div
          class="sc-Axmtr jNroUj"
        >
          <div
            class="sc-AxmLO jwnoix"
          >
            <div
              class="sc-fzozJi isqQvA"
            >
              <span
                class="sc-AxjAm bIGtDT"
                type="account"
              />
              <span
                class="sc-AxirZ iaUhNE"
              >
                Hanibal Buress
              </span>
            </div>
            <button
              class="sc-AxhCb inVoPr"
              type="tourDate"
            >
              <a
                class="sc-AxiKw gpmvAq"
                href=""
              >
                Get Tickets!
              </a>
            </button>
          </div>
          <h3
            class="sc-fzoLsD ifJBfs"
          >
            Baltimore, MD - Vincent Theater
          </h3>
          <div
            class="sc-fzpans cheDBt"
          >
            July 28th . 2020
          </div>
          <div
            class="sc-fzpans cheDBt"
          >
            Sunday . 8:30pm
          </div>
        </div>
      </div>
      <div
        class="sc-AxhUy kNWbAH"
      >
        <div
          class="sc-AxheI jzdBSq"
        >
          <svg
            class="sc-AxgMl bCKtDA"
            viewBox="0 0 512 512"
            x="0px"
            y="0px"
          >
            <path
              d="M256,0C115.2,0,0,115.2,0,256s115.2,256,256,256s256-115.2,256-256S396.8,0,256,0z M256,51.2
 c28.16,0,48.64,23.04,46.08,51.2L281.6,307.2h-51.2l-20.48-204.8C207.36,74.24,227.84,51.2,256,51.2z M256,460.8
 c-28.16,0-51.2-23.04-51.2-51.2c0-28.16,23.04-51.2,51.2-51.2s51.2,23.04,51.2,51.2C307.2,437.76,284.16,460.8,256,460.8z"
            />
          </svg>
          New Date!
        </div>
        <div
          class="sc-Axmtr jNroUj"
        >
          <div
            class="sc-AxmLO jwnoix"
          >
            <div
              class="sc-fzozJi isqQvA"
            >
              <span
                class="sc-AxjAm bIGtDT"
                type="account"
              />
              <span
                class="sc-AxirZ iaUhNE"
              >
                Hanibal Buress
              </span>
            </div>
            <button
              class="sc-AxhCb inVoPr"
              type="tourDate"
            >
              <a
                class="sc-AxiKw gpmvAq"
                href=""
              >
                Get Tickets!
              </a>
            </button>
          </div>
          <h3
            class="sc-fzoLsD ifJBfs"
          >
            Baltimore, MD - Vincent Theater
          </h3>
          <div
            class="sc-fzpans cheDBt"
          >
            July 28th . 2020
          </div>
          <div
            class="sc-fzpans cheDBt"
          >
            Sunday . 8:30pm
          </div>
        </div>
      </div>
      <div
        class="sc-AxhUy kNWbAH"
      >
        <div
          class="sc-AxheI jzdBSq"
        >
          <svg
            class="sc-AxgMl bCKtDA"
            viewBox="0 0 512 512"
            x="0px"
            y="0px"
          >
            <path
              d="M256,0C115.2,0,0,115.2,0,256s115.2,256,256,256s256-115.2,256-256S396.8,0,256,0z M256,51.2
 c28.16,0,48.64,23.04,46.08,51.2L281.6,307.2h-51.2l-20.48-204.8C207.36,74.24,227.84,51.2,256,51.2z M256,460.8
 c-28.16,0-51.2-23.04-51.2-51.2c0-28.16,23.04-51.2,51.2-51.2s51.2,23.04,51.2,51.2C307.2,437.76,284.16,460.8,256,460.8z"
            />
          </svg>
          New Date!
        </div>
        <div
          class="sc-Axmtr jNroUj"
        >
          <div
            class="sc-AxmLO jwnoix"
          >
            <div
              class="sc-fzozJi isqQvA"
            >
              <span
                class="sc-AxjAm bIGtDT"
                type="account"
              />
              <span
                class="sc-AxirZ iaUhNE"
              >
                Hanibal Buress
              </span>
            </div>
            <button
              class="sc-AxhCb inVoPr"
              type="tourDate"
            >
              <a
                class="sc-AxiKw gpmvAq"
                href=""
              >
                Get Tickets!
              </a>
            </button>
          </div>
          <h3
            class="sc-fzoLsD ifJBfs"
          >
            Baltimore, MD - Vincent Theater
          </h3>
          <div
            class="sc-fzpans cheDBt"
          >
            July 28th . 2020
          </div>
          <div
            class="sc-fzpans cheDBt"
          >
            Sunday . 8:30pm
          </div>
        </div>
      </div>
      <div
        class="sc-AxhUy kNWbAH"
      >
        <div
          class="sc-AxheI jzdBSq"
        >
          <svg
            class="sc-AxgMl bCKtDA"
            viewBox="0 0 512 512"
            x="0px"
            y="0px"
          >
            <path
              d="M256,0C115.2,0,0,115.2,0,256s115.2,256,256,256s256-115.2,256-256S396.8,0,256,0z M256,51.2
 c28.16,0,48.64,23.04,46.08,51.2L281.6,307.2h-51.2l-20.48-204.8C207.36,74.24,227.84,51.2,256,51.2z M256,460.8
 c-28.16,0-51.2-23.04-51.2-51.2c0-28.16,23.04-51.2,51.2-51.2s51.2,23.04,51.2,51.2C307.2,437.76,284.16,460.8,256,460.8z"
            />
          </svg>
          New Date!
        </div>
        <div
          class="sc-Axmtr jNroUj"
        >
          <div
            class="sc-AxmLO jwnoix"
          >
            <div
              class="sc-fzozJi isqQvA"
            >
              <span
                class="sc-AxjAm bIGtDT"
                type="account"
              />
              <span
                class="sc-AxirZ iaUhNE"
              >
                Hanibal Buress
              </span>
            </div>
            <button
              class="sc-AxhCb inVoPr"
              type="tourDate"
            >
              <a
                class="sc-AxiKw gpmvAq"
                href=""
              >
                Get Tickets!
              </a>
            </button>
          </div>
          <h3
            class="sc-fzoLsD ifJBfs"
          >
            Baltimore, MD - Vincent Theater
          </h3>
          <div
            class="sc-fzpans cheDBt"
          >
            July 28th . 2020
          </div>
          <div
            class="sc-fzpans cheDBt"
          >
            Sunday . 8:30pm
          </div>
        </div>
      </div>
      <div
        class="sc-AxhUy kNWbAH"
      >
        <div
          class="sc-AxheI jzdBSq"
        >
          <svg
            class="sc-AxgMl bCKtDA"
            viewBox="0 0 512 512"
            x="0px"
            y="0px"
          >
            <path
              d="M256,0C115.2,0,0,115.2,0,256s115.2,256,256,256s256-115.2,256-256S396.8,0,256,0z M256,51.2
 c28.16,0,48.64,23.04,46.08,51.2L281.6,307.2h-51.2l-20.48-204.8C207.36,74.24,227.84,51.2,256,51.2z M256,460.8
 c-28.16,0-51.2-23.04-51.2-51.2c0-28.16,23.04-51.2,51.2-51.2s51.2,23.04,51.2,51.2C307.2,437.76,284.16,460.8,256,460.8z"
            />
          </svg>
          New Date!
        </div>
        <div
          class="sc-Axmtr jNroUj"
        >
          <div
            class="sc-AxmLO jwnoix"
          >
            <div
              class="sc-fzozJi isqQvA"
            >
              <span
                class="sc-AxjAm bIGtDT"
                type="account"
              />
              <span
                class="sc-AxirZ iaUhNE"
              >
                Hanibal Buress
              </span>
            </div>
            <button
              class="sc-AxhCb inVoPr"
              type="tourDate"
            >
              <a
                class="sc-AxiKw gpmvAq"
                href=""
              >
                Get Tickets!
              </a>
            </button>
          </div>
          <h3
            class="sc-fzoLsD ifJBfs"
          >
            Baltimore, MD - Vincent Theater
          </h3>
          <div
            class="sc-fzpans cheDBt"
          >
            July 28th . 2020
          </div>
          <div
            class="sc-fzpans cheDBt"
          >
            Sunday . 8:30pm
          </div>
        </div>
      </div>
      <div
        class="sc-AxhUy kNWbAH"
      >
        <div
          class="sc-AxheI jzdBSq"
        >
          <svg
            class="sc-AxgMl bCKtDA"
            viewBox="0 0 512 512"
            x="0px"
            y="0px"
          >
            <path
              d="M256,0C115.2,0,0,115.2,0,256s115.2,256,256,256s256-115.2,256-256S396.8,0,256,0z M256,51.2
 c28.16,0,48.64,23.04,46.08,51.2L281.6,307.2h-51.2l-20.48-204.8C207.36,74.24,227.84,51.2,256,51.2z M256,460.8
 c-28.16,0-51.2-23.04-51.2-51.2c0-28.16,23.04-51.2,51.2-51.2s51.2,23.04,51.2,51.2C307.2,437.76,284.16,460.8,256,460.8z"
            />
          </svg>
          New Date!
        </div>
        <div
          class="sc-Axmtr jNroUj"
        >
          <div
            class="sc-AxmLO jwnoix"
          >
            <div
              class="sc-fzozJi isqQvA"
            >
              <span
                class="sc-AxjAm bIGtDT"
                type="account"
              />
              <span
                class="sc-AxirZ iaUhNE"
              >
                Hanibal Buress
              </span>
            </div>
            <button
              class="sc-AxhCb inVoPr"
              type="tourDate"
            >
              <a
                class="sc-AxiKw gpmvAq"
                href=""
              >
                Get Tickets!
              </a>
            </button>
          </div>
          <h3
            class="sc-fzoLsD ifJBfs"
          >
            Baltimore, MD - Vincent Theater
          </h3>
          <div
            class="sc-fzpans cheDBt"
          >
            July 28th . 2020
          </div>
          <div
            class="sc-fzpans cheDBt"
          >
            Sunday . 8:30pm
          </div>
        </div>
      </div>
      <div
        class="sc-AxhUy kNWbAH"
      >
        <div
          class="sc-AxheI jzdBSq"
        >
          <svg
            class="sc-AxgMl bCKtDA"
            viewBox="0 0 512 512"
            x="0px"
            y="0px"
          >
            <path
              d="M256,0C115.2,0,0,115.2,0,256s115.2,256,256,256s256-115.2,256-256S396.8,0,256,0z M256,51.2
 c28.16,0,48.64,23.04,46.08,51.2L281.6,307.2h-51.2l-20.48-204.8C207.36,74.24,227.84,51.2,256,51.2z M256,460.8
 c-28.16,0-51.2-23.04-51.2-51.2c0-28.16,23.04-51.2,51.2-51.2s51.2,23.04,51.2,51.2C307.2,437.76,284.16,460.8,256,460.8z"
            />
          </svg>
          New Date!
        </div>
        <div
          class="sc-Axmtr jNroUj"
        >
          <div
            class="sc-AxmLO jwnoix"
          >
            <div
              class="sc-fzozJi isqQvA"
            >
              <span
                class="sc-AxjAm bIGtDT"
                type="account"
              />
              <span
                class="sc-AxirZ iaUhNE"
              >
                Hanibal Buress
              </span>
            </div>
            <button
              class="sc-AxhCb inVoPr"
              type="tourDate"
            >
              <a
                class="sc-AxiKw gpmvAq"
                href=""
              >
                Get Tickets!
              </a>
            </button>
          </div>
          <h3
            class="sc-fzoLsD ifJBfs"
          >
            Baltimore, MD - Vincent Theater
          </h3>
          <div
            class="sc-fzpans cheDBt"
          >
            July 28th . 2020
          </div>
          <div
            class="sc-fzpans cheDBt"
          >
            Sunday . 8:30pm
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-AxhUy kNWbAH"
    >
      <div
        class="sc-AxheI jzdBSq"
      >
        <svg
          class="sc-AxgMl bCKtDA"
          viewBox="0 0 512 512"
          x="0px"
          y="0px"
        >
          <path
            d="M256,0C115.2,0,0,115.2,0,256s115.2,256,256,256s256-115.2,256-256S396.8,0,256,0z M256,51.2
 c28.16,0,48.64,23.04,46.08,51.2L281.6,307.2h-51.2l-20.48-204.8C207.36,74.24,227.84,51.2,256,51.2z M256,460.8
 c-28.16,0-51.2-23.04-51.2-51.2c0-28.16,23.04-51.2,51.2-51.2s51.2,23.04,51.2,51.2C307.2,437.76,284.16,460.8,256,460.8z"
          />
        </svg>
        New Date!
      </div>
      <div
        class="sc-Axmtr jNroUj"
      >
        <div
          class="sc-AxmLO jwnoix"
        >
          <div
            class="sc-fzozJi isqQvA"
          >
            <span
              class="sc-AxjAm bIGtDT"
              type="account"
            />
            <span
              class="sc-AxirZ iaUhNE"
            >
              Hanibal Buress
            </span>
          </div>
          <button
            class="sc-AxhCb inVoPr"
            type="tourDate"
          >
            <a
              class="sc-AxiKw gpmvAq"
              href=""
            >
              Get Tickets!
            </a>
          </button>
        </div>
        <h3
          class="sc-fzoLsD ifJBfs"
        >
          Baltimore, MD - Vincent Theater
        </h3>
        <div
          class="sc-fzpans cheDBt"
        >
          July 28th . 2020
        </div>
        <div
          class="sc-fzpans cheDBt"
        >
          Sunday . 8:30pm
        </div>
      </div>
    </div>
    <div
      class="sc-AxhUy kNWbAH"
    >
      <div
        class="sc-AxheI jzdBSq"
      >
        <svg
          class="sc-AxgMl bCKtDA"
          viewBox="0 0 512 512"
          x="0px"
          y="0px"
        >
          <path
            d="M256,0C115.2,0,0,115.2,0,256s115.2,256,256,256s256-115.2,256-256S396.8,0,256,0z M256,51.2
 c28.16,0,48.64,23.04,46.08,51.2L281.6,307.2h-51.2l-20.48-204.8C207.36,74.24,227.84,51.2,256,51.2z M256,460.8
 c-28.16,0-51.2-23.04-51.2-51.2c0-28.16,23.04-51.2,51.2-51.2s51.2,23.04,51.2,51.2C307.2,437.76,284.16,460.8,256,460.8z"
          />
        </svg>
        New Date!
      </div>
      <div
        class="sc-Axmtr jNroUj"
      >
        <div
          class="sc-AxmLO jwnoix"
        >
          <div
            class="sc-fzozJi isqQvA"
          >
            <span
              class="sc-AxjAm bIGtDT"
              type="account"
            />
            <span
              class="sc-AxirZ iaUhNE"
            >
              Hanibal Buress
            </span>
          </div>
          <button
            class="sc-AxhCb inVoPr"
            type="tourDate"
          >
            <a
              class="sc-AxiKw gpmvAq"
              href=""
            >
              Get Tickets!
            </a>
          </button>
        </div>
        <h3
          class="sc-fzoLsD ifJBfs"
        >
          Baltimore, MD - Vincent Theater
        </h3>
        <div
          class="sc-fzpans cheDBt"
        >
          July 28th . 2020
        </div>
        <div
          class="sc-fzpans cheDBt"
        >
          Sunday . 8:30pm
        </div>
      </div>
    </div>
    <div
      class="sc-AxhUy kNWbAH"
    >
      <div
        class="sc-AxheI jzdBSq"
      >
        <svg
          class="sc-AxgMl bCKtDA"
          viewBox="0 0 512 512"
          x="0px"
          y="0px"
        >
          <path
            d="M256,0C115.2,0,0,115.2,0,256s115.2,256,256,256s256-115.2,256-256S396.8,0,256,0z M256,51.2
 c28.16,0,48.64,23.04,46.08,51.2L281.6,307.2h-51.2l-20.48-204.8C207.36,74.24,227.84,51.2,256,51.2z M256,460.8
 c-28.16,0-51.2-23.04-51.2-51.2c0-28.16,23.04-51.2,51.2-51.2s51.2,23.04,51.2,51.2C307.2,437.76,284.16,460.8,256,460.8z"
          />
        </svg>
        New Date!
      </div>
      <div
        class="sc-Axmtr jNroUj"
      >
        <div
          class="sc-AxmLO jwnoix"
        >
          <div
            class="sc-fzozJi isqQvA"
          >
            <span
              class="sc-AxjAm bIGtDT"
              type="account"
            />
            <span
              class="sc-AxirZ iaUhNE"
            >
              Hanibal Buress
            </span>
          </div>
          <button
            class="sc-AxhCb inVoPr"
            type="tourDate"
          >
            <a
              class="sc-AxiKw gpmvAq"
              href=""
            >
              Get Tickets!
            </a>
          </button>
        </div>
        <h3
          class="sc-fzoLsD ifJBfs"
        >
          Baltimore, MD - Vincent Theater
        </h3>
        <div
          class="sc-fzpans cheDBt"
        >
          July 28th . 2020
        </div>
        <div
          class="sc-fzpans cheDBt"
        >
          Sunday . 8:30pm
        </div>
      </div>
    </div>
    <div
      class="sc-AxhUy kNWbAH"
    >
      <div
        class="sc-AxheI jzdBSq"
      >
        <svg
          class="sc-AxgMl bCKtDA"
          viewBox="0 0 512 512"
          x="0px"
          y="0px"
        >
          <path
            d="M256,0C115.2,0,0,115.2,0,256s115.2,256,256,256s256-115.2,256-256S396.8,0,256,0z M256,51.2
 c28.16,0,48.64,23.04,46.08,51.2L281.6,307.2h-51.2l-20.48-204.8C207.36,74.24,227.84,51.2,256,51.2z M256,460.8
 c-28.16,0-51.2-23.04-51.2-51.2c0-28.16,23.04-51.2,51.2-51.2s51.2,23.04,51.2,51.2C307.2,437.76,284.16,460.8,256,460.8z"
          />
        </svg>
        New Date!
      </div>
      <div
        class="sc-Axmtr jNroUj"
      >
        <div
          class="sc-AxmLO jwnoix"
        >
          <div
            class="sc-fzozJi isqQvA"
          >
            <span
              class="sc-AxjAm bIGtDT"
              type="account"
            />
            <span
              class="sc-AxirZ iaUhNE"
            >
              Hanibal Buress
            </span>
          </div>
          <button
            class="sc-AxhCb inVoPr"
            type="tourDate"
          >
            <a
              class="sc-AxiKw gpmvAq"
              href=""
            >
              Get Tickets!
            </a>
          </button>
        </div>
        <h3
          class="sc-fzoLsD ifJBfs"
        >
          Baltimore, MD - Vincent Theater
        </h3>
        <div
          class="sc-fzpans cheDBt"
        >
          July 28th . 2020
        </div>
        <div
          class="sc-fzpans cheDBt"
        >
          Sunday . 8:30pm
        </div>
      </div>
    </div>
    <div
      class="sc-AxhUy kNWbAH"
    >
      <div
        class="sc-AxheI jzdBSq"
      >
        <svg
          class="sc-AxgMl bCKtDA"
          viewBox="0 0 512 512"
          x="0px"
          y="0px"
        >
          <path
            d="M256,0C115.2,0,0,115.2,0,256s115.2,256,256,256s256-115.2,256-256S396.8,0,256,0z M256,51.2
 c28.16,0,48.64,23.04,46.08,51.2L281.6,307.2h-51.2l-20.48-204.8C207.36,74.24,227.84,51.2,256,51.2z M256,460.8
 c-28.16,0-51.2-23.04-51.2-51.2c0-28.16,23.04-51.2,51.2-51.2s51.2,23.04,51.2,51.2C307.2,437.76,284.16,460.8,256,460.8z"
          />
        </svg>
        New Date!
      </div>
      <div
        class="sc-Axmtr jNroUj"
      >
        <div
          class="sc-AxmLO jwnoix"
        >
          <div
            class="sc-fzozJi isqQvA"
          >
            <span
              class="sc-AxjAm bIGtDT"
              type="account"
            />
            <span
              class="sc-AxirZ iaUhNE"
            >
              Hanibal Buress
            </span>
          </div>
          <button
            class="sc-AxhCb inVoPr"
            type="tourDate"
          >
            <a
              class="sc-AxiKw gpmvAq"
              href=""
            >
              Get Tickets!
            </a>
          </button>
        </div>
        <h3
          class="sc-fzoLsD ifJBfs"
        >
          Baltimore, MD - Vincent Theater
        </h3>
        <div
          class="sc-fzpans cheDBt"
        >
          July 28th . 2020
        </div>
        <div
          class="sc-fzpans cheDBt"
        >
          Sunday . 8:30pm
        </div>
      </div>
    </div>
    <div
      class="sc-AxhUy kNWbAH"
    >
      <div
        class="sc-AxheI jzdBSq"
      >
        <svg
          class="sc-AxgMl bCKtDA"
          viewBox="0 0 512 512"
          x="0px"
          y="0px"
        >
          <path
            d="M256,0C115.2,0,0,115.2,0,256s115.2,256,256,256s256-115.2,256-256S396.8,0,256,0z M256,51.2
 c28.16,0,48.64,23.04,46.08,51.2L281.6,307.2h-51.2l-20.48-204.8C207.36,74.24,227.84,51.2,256,51.2z M256,460.8
 c-28.16,0-51.2-23.04-51.2-51.2c0-28.16,23.04-51.2,51.2-51.2s51.2,23.04,51.2,51.2C307.2,437.76,284.16,460.8,256,460.8z"
          />
        </svg>
        New Date!
      </div>
      <div
        class="sc-Axmtr jNroUj"
      >
        <div
          class="sc-AxmLO jwnoix"
        >
          <div
            class="sc-fzozJi isqQvA"
          >
            <span
              class="sc-AxjAm bIGtDT"
              type="account"
            />
            <span
              class="sc-AxirZ iaUhNE"
            >
              Hanibal Buress
            </span>
          </div>
          <button
            class="sc-AxhCb inVoPr"
            type="tourDate"
          >
            <a
              class="sc-AxiKw gpmvAq"
              href=""
            >
              Get Tickets!
            </a>
          </button>
        </div>
        <h3
          class="sc-fzoLsD ifJBfs"
        >
          Baltimore, MD - Vincent Theater
        </h3>
        <div
          class="sc-fzpans cheDBt"
        >
          July 28th . 2020
        </div>
        <div
          class="sc-fzpans cheDBt"
        >
          Sunday . 8:30pm
        </div>
      </div>
    </div>
    <div
      class="sc-AxhUy kNWbAH"
    >
      <div
        class="sc-AxheI jzdBSq"
      >
        <svg
          class="sc-AxgMl bCKtDA"
          viewBox="0 0 512 512"
          x="0px"
          y="0px"
        >
          <path
            d="M256,0C115.2,0,0,115.2,0,256s115.2,256,256,256s256-115.2,256-256S396.8,0,256,0z M256,51.2
 c28.16,0,48.64,23.04,46.08,51.2L281.6,307.2h-51.2l-20.48-204.8C207.36,74.24,227.84,51.2,256,51.2z M256,460.8
 c-28.16,0-51.2-23.04-51.2-51.2c0-28.16,23.04-51.2,51.2-51.2s51.2,23.04,51.2,51.2C307.2,437.76,284.16,460.8,256,460.8z"
          />
        </svg>
        New Date!
      </div>
      <div
        class="sc-Axmtr jNroUj"
      >
        <div
          class="sc-AxmLO jwnoix"
        >
          <div
            class="sc-fzozJi isqQvA"
          >
            <span
              class="sc-AxjAm bIGtDT"
              type="account"
            />
            <span
              class="sc-AxirZ iaUhNE"
            >
              Hanibal Buress
            </span>
          </div>
          <button
            class="sc-AxhCb inVoPr"
            type="tourDate"
          >
            <a
              class="sc-AxiKw gpmvAq"
              href=""
            >
              Get Tickets!
            </a>
          </button>
        </div>
        <h3
          class="sc-fzoLsD ifJBfs"
        >
          Baltimore, MD - Vincent Theater
        </h3>
        <div
          class="sc-fzpans cheDBt"
        >
          July 28th . 2020
        </div>
        <div
          class="sc-fzpans cheDBt"
        >
          Sunday . 8:30pm
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
